API.md

```diff
@@ -22,35 +22,46 @@ are augmented as follows:
-              function with signature `function(context, options, callback)` (the compiled async template) and `callback` has the
-              signature `function(err, rendered)`.
+              `compileMode` settings (see below).
+                - If `compileMode` is `'sync'`, the signature is `compile(src, options)`, where:
+                    - `src` - the template as a string.
+                    - `options` - the `compileOptions` object.
+                    - the return value is a function with signature `function(context, options)` (the compiled sync template), and the method is allowed to throw errors, where:
+                        - `context` - the object containing values to be rendered in the template.
+                        - `options` - the `runtimeOptions` object.
+                - If `compileMode` is `'async'`, the signature is `compile(src, options, next)` where:
+                    - `src` - the template as a string.
+                    - `options` - the `compileOptions` object.
+                    - `next` - has the signature `function(err, compiled)`.
+                        - `err` - is `null` or `Error`.
+                        - `compiled` is a function with signature `function(context, options, callback)` (the compiled async template) where:
+                            - `context` - the object containing values to be rendered in the template.
+                            - `options` - the `runtimeOptions` object.
+                            - `callback` has the signature `function(err, rendered)`, where:
+                                - `err` - is `null` or an `Error`.
+                                - `rendered` - is a string or `undefined`.
             - `prepare(config, next)` - initializes additional engine state.
               The `config` object is the engine configuration object allowing updates to be made.
               This is useful for engines like Nunjucks that rely on additional state for rendering.
               `next` has the signature `function(err)`.
             - `registerPartial(name, src)` - registers a partial for use during template rendering.
-              The `name` is the partial path that templates should use to reference the partial and
+              The `name` is the partial path (string) that templates should use to reference the partial and
               `src` is the uncompiled template string for the partial.
             - `registerHelper(name, helper)` - registers a helper for use during template rendering.
-              The `name` is the name that templates should use to reference the helper and `helper`
-              is the function that will be invoked when the helper is called.
+              The `name` is the name (string) that templates should use to reference the helper and `helper`
+              is the function that will be invoked when the helper is called, with signature: `(...args: any[]): void`.
```

Comment:
Can we omit the func signature for helpers? I don't think it adds much value since it accepts whatever you pass it in the templates